astropy/io/ascii/cds.py

```diff
@@ -379,7 +380,7 @@ def read(self, table):
             with suppress(TypeError):
                 # For strings only
                 if os.linesep not in table + "":
```

Comment:
@taldcroft just out of pure curiosity, what is the purpose of the + "" here?

Reply:
I assume this is to trigger a `TypeError` if `table` isn't a string ?

Reply:
Exactly what @neutrinoceros said. This is a very old idiom to check for string-like (somewhat like a protocol, asking if it supports appending a `str`) instead of an explicit `str` type check.

What's the best way to check for string-like? Or does `isinstance(table, str)` cover all realistic use cases?

Reply:
From `BaseReader.read` in core.py below, which uses the same construct, the only other inputs are file-like or list of strings (the table lines), so they should all be covered.